File: internal/app/visualiseRepository/visualiseRepository.pl
#!/usr/bin/perl

my $rootRepositoryDirectory = "../../../";

system("cloc-1.78.exe", $rootRepositoryDirectory, "--exclude-dir=vendor");  # https://github.com/AlDanial/cloc
system("gource", "--load-config", "gource.config");                         # https://github.com/acaudwell/Gource

File: internal/app/visualiseRepository/visualiseRepository.py
# (c) 2020, Australian Rivers Institute
# Author: Lindsay Bradford

# http://cloc.sourceforge.net/

import subprocess


def main():
    showCountOfLinesOfCode()
    visualiseRepositoryViaGource()


def showCountOfLinesOfCode():
    rootRepositoryDirectory = "../../../"
    clocArray = ['cloc-1.88.exe', rootRepositoryDirectory]

    clocOutput = subprocess.run(clocArray, capture_output=True, text=True)
    print(clocOutput.stdout)


def visualiseRepositoryViaGource():
    gourceArray = ['gource', '--load-config', 'gource.config']
    subprocess.run(gourceArray, shell=True)


if __name__ == '__main__':
    main()